### ISSUE TITLE
Metalness implementation seems incorrect

### ISSUE THREAD

**Issue by gkjohnson** [gkjohnson/three-gpu-pathtracer#76]
**TODO**
- [ ] Check existing function correctness
- [ ] Compare against existing implementations
- [ ] Understand why roughness grows so quickly
- [ ] Non metal has no specular
- [ ] Squaring the roughness improves comparison
- [x] ~Ensure textures are in correct color space~

See straps that should be dark gray and yellow (and for some reason have a high metalness value)

![image](https://user-images.githubusercontent.com/734200/161447456-f27adfbc-b3ab-4105-8aa0-66e11c56585e.png)

**Comment by gkjohnson** [comment]
Is possible this is a normal map / texture wrapping issue? Are UVs within [0, 1]?

**Comment by gkjohnson** [comment]
This can somewhat be emulated with the material ball when setting the color to be fairly dark, roughness high, and then ramping the metalness up and down. It seems to get really bright in the metalness mid range and entirely dark at the ends. Perhaps theres an issue with the GGX functions?

**Comment by gkjohnson** [comment]
Related: #221

**Comment by gkjohnson** [comment]
See metalness (or maybe just glossiness) issues in the metal spheres comparison:

![image](https://user-images.githubusercontent.com/734200/193441371-f09b6c00-e4c7-4553-8cca-9bf16b2dd7b2.png)

**Comment by gkjohnson** [comment]
Squaring the roughness improves the falloff and make it look closer to the comparison though the "smoothest" setting still has low relflection. Metal may be the same. Perhaps the textures are being read in the wrong color space? Or are stored in the wrong color space?

<img width="400" alt="image" src="https://user-images.githubusercontent.com/734200/193466321-4cf3819f-a2da-4806-b73f-c54674fa3348.png">

**Comment by gkjohnson** [comment]
Glancing fresnel looks incorrect:

<img width="1562" alt="image" src="https://user-images.githubusercontent.com/734200/193477397-0f7d7286-6bb7-448f-b3f0-d38bbccef72d.png">

**Comment by gkjohnson** [comment]
Closing in favor of #264

**Comment by gkjohnson** [comment]
Aviation Helmet after #315:

<img width="614" alt="image" src="https://user-images.githubusercontent.com/734200/206835816-a5cea0ff-1aba-411a-9ce6-7511bfea5013.png">

And Specular Test from Model Viewer:

<img width="771" alt="image" src="https://user-images.githubusercontent.com/734200/206835845-02f74432-e030-4777-bad0-71a4f3fa9081.png">
